pulp_container/app/token_verification.py

```diff
@@ -80,13 +81,18 @@ def authenticate(self, request):
             return (AnonymousUser, None)
 
         try:
-            return super().authenticate(request)
+            result = super().authenticate(request)
```

Comment:
`None` is returned only when https://github.com/encode/django-rest-framework/blob/9864c47018238dd9f8477df0e4ab2ef62a24fd2b/rest_framework/authentication.py#L66 this is evaluated to `True`. We exploit this check.

Reply:
That sounds brittle. Even with this explanation it's hard to understand. You should have a comment at the very least. Is the "empty username and password" working at all with remote auth?

Reply:
What does the empty username and password mean in this context? If there is no `Remote_User` passed with the request, and no Authorization header is present, Pulp consider this request being anonymous, allowing standard pull operations for public repositories.

Reply:
What i mean is: When remote auth is configured, and the reverse proxy terminates the authentication process, does it also accept empty for anonymous?

Reply:
Yes, `RemoteUserRegistryAuthentication().authenticate(request)` returns `None` if there is no `Remote_User` header set.

Reply:
I renamed `result` into a more explicit value: `authenticated_user`.